src/platform/posix/posix_clock.c

```diff
@@ -19,13 +19,45 @@
 
 #ifndef NNG_USE_GETTIMEOFDAY
 
-// Use POSIX realtime stuff
+#if __APPLE__
+#include <mach/mach.h>
+#include <mach/mach_time.h>
+#endif
+
+// Use POSIX realtime stuff                                                          
 nni_time
 nni_clock(void)
 {
 	struct timespec ts;
 	nni_time        msec;
 
+#if __APPLE__
+
+	int supported = 0;
+
+#if defined(MAC_OS_X_VERSION_10_12) && __has_builtin(__builtin_available)
+	if (__builtin_available(macOS 10.12, *)) {
```

Comment:
is it possible (since that evaluates to a boolean type value ) to just use it inline ?

```
bool has_clock_gettime = __builtin_available(macOS 10.12, *);
```

Reply:
Nope, `__builtin_available()` can only be used inside an `if` (I lost a reference for that and it's too late here to look it up again).

Reply:
bummer